.github/workflows/release.yaml

```diff
@@ -1,17 +1,17 @@
-name: Docs build
-on:
--   push
+name: Release
+
+on: [push]
 
 jobs:
-    build:
+    build_and_deploy_docs:
```

Comment:
@nilskre didn't we want to build the docs always, not just on a new release?

Reply:
maybe a separate yaml for that would be easier to understand / maintain.

Reply:
Yes, we can also build the docs always. I think for the deployment of the docs/artifact there are still some changes to be done within the next PR. So we could have a chat about this and then include the changes in the next PR. Would this be alright?

Reply:
> maybe a separate yaml for that would be easier to understand / maintain.

Yes, I totally agree. So then three yaml files:
- ci_cd.yaml: test and code quality
- docs.yaml: build (and publish?) docs
- release.yaml: build and publish artifact

Are the names alright like this or could we improve understanding by other names?

Reply:
> include the changes in the next PR. 

Sure, as long as all checks pass in this PR. I am now thinking that it would probably be easier to check the deploy functionality if we directed this PR at a feature branch (like `pyopensci`) instead of `main` directly. I have redirected the PR; we can merge it with lower risk.

Reply:
> Are the names alright

Those are fine, and yes, publishing should also happen in the docs workflow.

Reply:
To clarify the build and deployment of the docs:
- build the docs -> on every commit

Regarding deployment, I could imagine three scenarios:
[current] 1) Deploy the docs on every tag on the main branch and have one version of the docs: stable (state of the latest tag)
2) Deploy the docs on every commit on the main branch and have two versions of the docs: stable (state of the latest tag) and latest (latest commit on the main branch).
3) Deploy the docs on every commit on the main branch and have multiple versions of the docs: vX.X.X (docs for every version/tag), stable (state of the latest tag), and latest (latest commit on the main branch).

For now, I think option 1 is totally sufficient. But I could imagine, that possibility 2 or 3 could maybe be something for a future improvement?

So for now continue with option 1 and create an issue to discuss options 2 and 3. Would this be alright for you?

Reply:
Yes, I would keep it simple for now.

Reply:
Great, then this is something to keep in mind when reaching the first major release of BioCypher.